mobile_verifier/src/coverage.rs

```diff
@@ -496,10 +498,44 @@ pub async fn clear_coverage_objects(
     Ok(())
 }
 
+#[derive(Default, Debug)]
+pub struct CoveredIndoorHex {
+    multiplier: u32,
+    coverage: BTreeMap<SignalLevel, BinaryHeap<IndoorCoverageLevel>>,
+}
+
+impl CoveredIndoorHex {
+    pub fn add_coverage(
+        &mut self,
+        multiplier: u32,
+        signal_level: SignalLevel,
+        coverage_level: IndoorCoverageLevel,
+    ) {
+        self.multiplier = multiplier;
+        self.coverage
+            .entry(signal_level)
+            .or_default()
+            .push(coverage_level)
+    }
+}
```

Comment:
It doesn't make sense to me that you are setting the multiplier in the same place where you add a coverage level. 

```rust
impl CoveredIndoorHex {
    pub fn new(boosted_hex_multiplier: u32) -> Self { ... }
}
```

Reply:
Good point, this goes back to an earlier implementation when i was applying the multiplier earlier in the flow. I can remove it from here entirely and use only at the point of going to coverage rewards